cpp/src/quantiles/quantiles.cu

```diff
@@ -100,44 +63,19 @@ namespace{ // anonymous
       thrust::sort(rmm::exec_policy(stream)->on(stream), devarr, devarr+n);
     }
 
-    QuantileIndex qi(n, quant);
+    CUDF_EXPECTS(interpolation < cudf::N_QUANT_METHODS &&

```

Comment:
I don't think the compiler will let you pass an interpolation other than one of the members of the enum. So it should be sufficient to check 

```suggestion
    CUDF_EXPECTS(interpolation != cudf::N_QUANT_METHODS, "Invalid quantile interpolation method");
```

Reply:
It doesn't seem to stop me https://onlinegdb.com/By2_Vx4rS
We shouldn't do this but never hurts to be safe.